Sources/SwiftGRPCNIO/CallHandlers/ClientStreamingCallHandler.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+import SwiftProtobuf
+import NIO
+import NIOHTTP1
+
+// Handles client-streaming calls. Forwards incoming messages and end-of-stream events to the observer block.
+// The observer block is implemented by the framework user and fulfills `context.responsePromise` when done.
+public class ClientStreamingCallHandler<RequestMessage: Message, ResponseMessage: Message>: BaseCallHandler<RequestMessage, ResponseMessage> {
+  public typealias EventObserver = (StreamEvent<RequestMessage>) -> Void
+  fileprivate var eventObserver: EventLoopFuture<EventObserver>?
```

Comment:
I'll avoid commenting on all these cases, but we should clean up the `fileprivate --> private` and force unwraps throughout this PR

Reply:
Done. Let me know in case I missed any.